### PR TITLE
Fix compiler warning unicode chars

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: https://github.com/Azure/azure-sdk-for-c/issues/1087

### CODE REVIEW

File: sdk/src/azure/core/az_http_header_validation_private.h [hunk 1]
```diff
@@ -41,7 +41,7 @@ static const uint8_t az_http_valid_token[256] = {
   0, // 16 ->  data link escape
   0, // 17 ->  device control 1
   0, // 18 ->  device control 2
-  0, // 19 ->  device control 3
+  0, // 19 ->  device control 3
```
Comment (ahsonkhan) [inline sdk/src/azure/core/az_http_header_validation_private.h:44]: Lol, if anyone is wondering about what these ghostly changes are. We went from non-ascii-compliant non-breaking space character to the regular, 0x20 whitespace character :)

File: sdk/src/azure/core/az_http_header_validation_private.h [hunk 2]
```diff
@@ -56,18 +56,18 @@ static const uint8_t az_http_valid_token[256] = {
...
   '&', // 38 ->  &
-  '\'', // 39 ->  ‘
+  '\'', // 39 -> '
   0, // 40 ->  (
   0, // 41 ->  )
   '*', // 42 ->  *
   '+', // 43 ->  +
   0, // 44 ->  ,
-  '-', // 45 ->  –
+  '-', // 45 ->  -
```
Comment (ahsonkhan) [inline sdk/src/azure/core/az_http_header_validation_private.h:70]: @vhvb1989 - did you verify that we don't have any other non-ASCII characters in the file?

**Review by ahsonkhan — APPROVED** [review]
Looks good.

File: sdk/src/azure/core/az_http_header_validation_private.h (hunk 2 above)
Comment (vhvb1989) [reply sdk/src/azure/core/az_http_header_validation_private.h:70]: I did. 
in VSCode we can re-open file with an specific encoding.  I checked there is no other weird encoding

Comment (ahsonkhan) [reply sdk/src/azure/core/az_http_header_validation_private.h:70]: Also, this returned no results. So all is good:

> grep --color='auto' -P -n "[\x80-\xFF]" sdk/src/azure/core/az_http_header_validation_private.h

**Pull request merged**